### DISCUSSION TITLE
Monitor and signal the current consumption of a device using Tasmota

### DISCUSSION THREAD

**Discussion by c0nne1504683nf** [arendst/Tasmota#11977] [Q&A] [answered]
Hello All,

unfortunately, I'm fairly new to the "club" of Tasmota users.

I own a wireless socket GOSUND SP112 and would like to use it to monitor the on/off state of a compressor by measuring the power consumption and sending a message via MQTT each time the motor starts up or shuts down.

The SP112 is already flashed with Tasmota, also the wifi- and MQTT configuration is done. 

I still have to calibrate the current and voltage measurement, for this I am currently missing an adapter for the multimeter as a reference instrument. But I think I know how the calibration is done in principle.

I have also read a lot, but unfortunately not understood everything. I think I need to create a rule and install it on the SP112. As trigger in my case one takes probably the command PowerDelta?! But this is all theory for now.

Unfortunately, I am now somewhat helpless as far as the practical implementation is concerned. What is the correct syntax for the rule and what is the easiest way to get it into the SP112?

I would appreciate any kind of help to get started.

Thanks!

Best,
c0nne1504683nf

**Comment by barbudor** [comment, marked as answer]
With an energy monitor device like SP112, the default behavior is for Tasmota to report the power consumption on regular interval called Teleperiod. You can change this period with the command of the same name.
Then as you discovered, the command PowerDelta can set Tasmota to send additional reports when the consumption has changed buy either a fixed amount or a percentage.

Those are convenient to track the power consumption such as writing the consumption on a database and display graphs.

If you are only interested in detecting when the device is switched on and off, the simplest is to use a rule that will compare the power with a threshold. For that a trigger like `on ENERGY#Power>20 do ...` and `on ENERGY#Power<20 do ...` would be sufficient. You can also replace the value by `%mem1%` and store the threshold value into variable Mem1 with command `Mem1 20` which makes it easier to adjust. 
To avoid the rule to be triggered at each measurement but only when the threshold is crossed, use the "one-shot" mode.

So my suggestion would be:
```
mem1 20
rule1 on ENERGY#Power>%mem1% do publish stat/devicetopic/OPERATING ON endon 
rule1 +on ENERGY#Power<%mem1% do publish stat/devicetopic/OPERATING OFF endon 
rule1 5
rule1 1
```
You can of course replace the publish action by something else

**Comment by c0nne1504683nf** [comment]
Hi barbudor,

thank you very much for your valuable advice!!!

Today I first calibrated my SP112 and then entered the rule setup commands in the console of WebUI. I could immediately see the response of the device in the log. It works great!

I assume that the rule settings are saved so that they are still effective after a power-on reset?

Now for the compressor monitoring application all I still need is an MQTT broker and a simple way to write the MQTT messages together with timestamps into a file. I think something like an InfluxDB is completely overkill for this purpose?!

By the way: Is it actually possible to have two instances of Tasmota communicate with each other completely without a MQTT broker? For another project I'm thinking of an ESP developer board (e.g. WEMOS D1 mini) with a float switch and another wifi socket to switch a water pump on and off depending on the filling level. 

A MQTT broker would actually be a useless overhead for this simple task. However, Tasmota obviously has to send its messages to a MQTT broker?!

Once again, thanks you for your help!

Best,
c0nne1504683nf

**Comment by barbudor** [comment]
Rules are stored in flash

Indeed MQTT is the best to get messages out of Tasmota
There is possibility of using Websend to make http Get request but error processing is not easy. Mqtt is more reliable. 

MQTT is only a post office, you need someone to process the message. If you not yet à home automation system, you may go for Nodered which can process the MQTT messages.
InfluxDB/Grafana is overkill but that's always an opportunity to learn new things. 
Otherwise Nodered can write to a text file, or you can use any other database.
